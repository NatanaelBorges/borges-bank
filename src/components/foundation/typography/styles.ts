import styled, { css, DefaultTheme } from 'styled-components';
import { TypographyProps } from '.';

const textSize = {
  h1: (theme: DefaultTheme) => css`
    font-size: ${theme.font.sizes.heading.huge};
  `,
  h2: (theme: DefaultTheme) => css`
    font-size: ${theme.font.sizes.heading.large};
  `,
  h3: (theme: DefaultTheme) => css`
    font-size: ${theme.font.sizes.heading.big};
  `,
  h4: (theme: DefaultTheme) => css`
    font-size: ${theme.font.sizes.heading.medium};
  `,
  h5: (theme: DefaultTheme) => css`
    font-size: ${theme.font.sizes.heading.small};
  `,
  h6: (theme: DefaultTheme) => css`
    font-size: ${theme.font.sizes.heading.xsmall};
  `,
};

const textSizeParagraph = {
  xsmall: (theme: DefaultTheme) => css`
    font-size: ${theme.font.sizes.paragraph.xsmall};
  `,
  small: (theme: DefaultTheme) => css`
    font-size: ${theme.font.sizes.paragraph.small};
  `,
  medium: (theme: DefaultTheme) => css`
    font-size: ${theme.font.sizes.paragraph.medium};
  `,
  big: (theme: DefaultTheme) => css`
    font-size: ${theme.font.sizes.paragraph.big};
  `,
};

const textCase = (uppercase: boolean) => css`
  text-transform: ${uppercase ? 'uppercase' : 'none'};
`;

const textItalic = (italic: boolean) => css`
  font-style: ${italic ? 'italic' : 'none'};
`;

const textUnderline = (underline: boolean) => css`
  text-decoration: ${underline ? 'underline' : 'none'};
`;

const textBold = (bold: boolean) => css`
  font-weight: ${bold ? 'bold' : 'normal'};
`;

const textColor = {
  primary: (theme: DefaultTheme) => css`
    color: ${theme.colors.primary};
  `,
  secondary: (theme: DefaultTheme) => css`
    color: ${theme.colors.secondary};
  `,
  white: (theme: DefaultTheme) => css`
    color: ${theme.colors.white};
  `,
  black: (theme: DefaultTheme) => css`
    color: ${theme.colors.black};
  `,
};

const textDisplay = {
  display1: (theme: DefaultTheme) => css`
    font-size: ${theme.font.sizes.display.huge};
  `,
  display2: (theme: DefaultTheme) => css`
    font-size: ${theme.font.sizes.display.large};
  `,
  display3: (theme: DefaultTheme) => css`
    font-size: ${theme.font.sizes.display.big};
  `,
  display4: (theme: DefaultTheme) => css`
    font-size: ${theme.font.sizes.display.medium};
  `,
  display5: (theme: DefaultTheme) => css`
    font-size: ${theme.font.sizes.display.small};
  `,
  display6: (theme: DefaultTheme) => css`
    font-size: ${theme.font.sizes.display.xsmall};
  `,
};

export const Text = styled.h1<TypographyProps>`
  ${({
    theme,
    align,
    italic,
    uppercase,
    as,
    display,
    size,
    underline,
    bold,
    color,
  }) => css`
    text-align: ${align};
    ${textCase(uppercase)};
    ${textColor[color](theme)};
    ${textUnderline(underline)};
    ${textItalic(italic)};
    ${textBold(bold)};
    ${as === 'p' ? textSizeParagraph[size](theme) : textSize[as](theme)};
    ${display !== 'none' ? textDisplay[display](theme) : 'none'};
  `}
`;
