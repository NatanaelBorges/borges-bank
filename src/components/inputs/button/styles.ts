import styled, { css } from 'styled-components';
import { Link } from 'react-scroll';
import { ButtonProps } from '.';

export const Button = styled.button<ButtonProps>`
  ${({ theme, fontBig, big, primary, dark }) => css`
    border-radius: 50px;
    background: ${primary ? theme.colors.primary : theme.colors.dark};
    white-space: nowrap;
    padding: ${big ? '14px  48px' : '12px 30px'};
    color: ${dark ? theme.colors.dark : theme.colors.white};
    font-size: ${fontBig ? '20px' : '16px'};
    outline: none;
    border: none;
    cursor: pointer;
    display: flex;
    justify-content: center;
    align-items: center;
    transition: all 0.2s ease-in-out;

    &:hover {
      transition: all 0.2s ease-in-out;
      background: ${primary ? theme.colors.white : theme.colors.primary};
    }
  `}
`;

export const href = styled(Link)``;
