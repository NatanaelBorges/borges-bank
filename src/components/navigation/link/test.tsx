import { screen } from '@testing-library/react';
import { renderTheme } from '../../../styles/render-theme';
import { Link } from '.';

describe('<MenuLink />', () => {
  const onPressEvent = jest.fn();
  onPressEvent.mockReturnValue('Link on press invoked');

  it('should render a link', () => {
    renderTheme(
      <Link link="http://localhost" onPress={onPressEvent}>
        Children
      </Link>,
    );
    expect(screen.getByRole('link', { name: 'Children' })).toHaveAttribute(
      'target',
      '_self',
    );
  });

  it('should render a internal link', () => {
    renderTheme(
      <Link link="/localhost" onPress={onPressEvent}>
        Children
      </Link>,
    );
    expect(screen.getByRole('link', { name: 'Children' })).toHaveAttribute(
      'target',
      '_self',
    );
  });

  it('should render open in a new tab', () => {
    renderTheme(
      <Link link="http://localhost" newTab={true} onPress={onPressEvent}>
        Children
      </Link>,
    );
    expect(screen.getByRole('link', { name: 'Children' })).toHaveAttribute(
      'target',
      '_blank',
    );
  });

  it('should render open in a new tab', () => {
    const { container } = renderTheme(
      <Link link="http://localhost" newTab={false} onPress={onPressEvent}>
        Children
      </Link>,
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
