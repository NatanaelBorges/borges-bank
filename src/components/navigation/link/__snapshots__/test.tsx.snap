// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MenuLink /> should render open in a new tab 1`] = `
<a
  class=""
  href="http://localhost"
  target="_self"
>
  Children
</a>
`;
